lib/PluginBase.js

```diff
@@ -36,11 +36,11 @@ class PluginBase {
      * Method for Plugin developer to initialize his Plugin
      *
      * @param {Record<string, any>} pluginConfig plugin configuration from config files
-     * @param {import("@iobroker/plugin-base/types").InitCallback} callback Will be called when done. On err or `initSuccessful === false` the plugin instance will be discarded.
+     * @return {Promise<void>} resolves if init was successful else rejects
      */
-    init(pluginConfig, callback) {
+    init(pluginConfig) {
```

Comment:
make async? ... also for the other methods ... then simply throw

Reply:
Why? They are returning `Promise<void>` On type level how you achieve this on implementation is up to code of specific plugin, there it will likely be `async`. See https://github.com/ioBroker/plugin-base/pull/126/files#diff-91cd6348c12238c4c2afe46f71253579dbc43ddb1db8872b76bff80ff25025e2R34 for type level

Reply:
yes... just personal preference and "the same as in other classes" ... I hate "new Promise.reject" constructions in the meantime :-)

Reply:
Normally we have eslint rules which complain if you have async function without doing anything async thus I prefer new Promise for such scenarios 😀

Reply:
lol ... ok ... strange ... the throw normally suppreses that ... but all good